packages/idempotency/src/persistence/IdempotencyRecord.ts

```diff
@@ -1,15 +1,31 @@
-import { IdempotencyRecordStatus } from '../types';
 import type { IdempotencyRecordOptions } from '../types';
+import { IdempotencyRecordStatus } from '../types';
 import { IdempotencyInvalidStatusError } from '../Exceptions';
 
 /**
- * Class representing an idempotency record
+ * Class representing an idempotency record.
+ * The properties of this class will be reflected in the persistance layer.
  */
 class IdempotencyRecord {
+  /**
+   * The expiry timestamp of the record in milliseconds UTC.
+   */
   public expiryTimestamp?: number;
+  /**
+   * The idempotency key of the record that is used to identify the record.
+   */
   public idempotencyKey: string;
+  /**
+   * The expiry timestamp of the in progress record in milliseconds UTC.
+   */
   public inProgressExpiryTimestamp?: number;
+  /**
+   * The hash of the payload of the request, used for comparing requests.
+   */
   public payloadHash?: string;
+  /**
+   * The response data of the request, this will be returned if the payload hash matches.
+   */
   public responseData?: Record<string, unknown>;
   private status: IdempotencyRecordStatus;
```

Comment:
Can we add a comment here like the others & maybe link to `IdempotencyRecordStatus`?